src/main/java/spoon/reflect/visitor/PrinterHelper.java

```diff
@@ -231,10 +231,9 @@ public PrinterHelper adjustStartPosition(CtElement e) {
 	}
 
 	/**
-	 * If the environment in which spoon is executed tries to preserve the original line number and the the passed
-	 * indexes of CtElement e are known then it will generate new lines
-	 * @param e is a {@link CtElement} on which new lines are going to be generated
-	 * @return {@link PrinterHelper}
+	 * writes as many newlines as needed for the current line to match the end line of the passed element
+	 * @param e is a CtElement whose line number will be preserved by adding newlines
+	 * @return PrinterHelper
```

Comment:
I think you misinterpreted my feedback about referring to types: just writing e.g. `CtElement` is less informative than `{@link CtElment}`, and it _still_ refers explicitly to the type. Look for example at [this standard library Javadoc](https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#add(int,%20E)). The Javadoc describes the semantic meaning of the values, not their types. Note also that the type of each parameter is very evident in the Javadoc as they appear in the signature, so unless it actually helps understand something, I would omit the explicit types.

 What I meant was something like this:

```java
// no; refers explicitly to the type, which is already specified in the method header
* @param e is a {@link CtElement} whose line number will be preserved by adding newlines

// no; also refers explicitly to the type, but there will be no clickable link in the Javadoc HTML
* @param e is a CtElement whose line number will be preserved by adding newlines

// yes; does not refer explicitly to the type, we just say "element", which is more general
* @param e element whose line number will be preserved by adding newlines
```

To be clear, if you need to refer to the type, use `@link` tags. If you don't need to refer to the type, describe the value in natural language without the type name.

Reply:
Thanks for sharing the example and clearing things up, for now on I will just ask if something isn't clear to me.

Reply:
> for now on I will just ask if something isn't clear to me.

Very good!